### ISSUE TITLE
Training CIF

### ISSUE THREAD

**Issue by wh1t3tea** [backspacetg/simul_whisper#13]
Hello, I am trying to train CIF layer for whisper-turbo, however I faced some troubles. As mentioned in another issue about training CIF, your training objective :  
```python 
features = whisper.encoder(mel)
alphas = cif_layer(features)
alphas = torch.sigmoid(alphas).squeeze(-1)
alphas = alphas * (~padding_mask.squeeze(1)).float()
loss = rmse(alphas.sum(-1), number_of_words_in_sequence)
```
I reproduced this and my losses decreasing on train and validation to values under 1.0, however when i try to inference Simul Whisper pipeline with this layer CIF never fires. Can you help me to understand what I am doing wrong please.

Or maybe I should mask encoder outputs before cif layer?
